### ISSUE TITLE
Swap thumbs on range slider when thumbs are overlapping

### ISSUE THREAD

**Issue by benjavicente** [mui/base-ui#1220]
Coming from `mui-base`, `base-ui-components` seems to lack the correct thumb placement (will be fixed with https://github.com/mui/base-ui/pull/1175) and the ability to swap thumbs.  While swapping values does not follow the [Slider WAI-ARIA design pattern](https://www.w3.org/WAI/ARIA/apg/patterns/slider-multithumb) (see [relevant RadixUI discussion about that](https://github.com/radix-ui/primitives/issues/2247#issuecomment-2299363019)), swapping might provide better UX, specially when thumbs are overlapping.

Where is a comparison with`base-ui-components`  vs  `mui-base`. In the base ui example, since the clicks happen on the left side of the thumb, the left thumb is selected, even thought there is an intent to slide to the right.

https://github.com/user-attachments/assets/7ae03807-237f-4d37-95a0-4122566ae882

https://github.com/user-attachments/assets/3751dec9-bd28-4dc6-9633-8b7080dcab54

So, it could be great to have the ability to enable always thumb swaping like in `mui-base`, and would be amazing if this library manages to include a threshold of when it should overlap. I couldn't find a library that does this last behavior 🤔

**Comment by mj12albert** [comment]
Related to https://github.com/mui/base-ui/issues/420

@benjavicente We decided to not allow thumbs to cross each other for two main reasons:
1. In the legacy package, the thumbs never actually swap positions. When they appear to "swap", focus just magically moves to the other thumb which is not great for accessibility.
2. Solving the above means actually reordering the thumb components/elements in the DOM as they are dragged around, and will require a more complicated API - e.g. using a function child to render thumbs, instead of manually composing them one by one - which is also something we want to avoid at the moment

> would be amazing if this library manages to include a threshold of when it should overlap. I couldn't find a library that does this last behavior 🤔

There is a `minStepsBetweenValues` prop that you can use to enforce the min distance between thumbs and prevent overlap (denominated in no. of steps) in a range slider, and will help avoid landing on a "range" like `10-10`.
(I think radix has a similar prop)

```jsx
<Slider.Root value={[10, 20]} step={1} minStepsBetweenValues={1} />
```

**Comment by mj12albert** [comment]
It will be a huge amount of work in figuring out how to overcome the accessibility challenges of doing this, so we won't do this at least in the near future

We can still label it "waiting for upvotes" so the community can vote for this. If you want to see this implemented, please add a 👍🏼 reaction to the issue's description!

**Comment by Bandit** [comment]
If the following is true:

1. Need a slider with a min and max range
2. Need to allow a single value to be selected

Then not allowing the thumbs to "swap" is a UX nightmare, far greater than the corresponding "accessibility nightmare"

**Comment by mj12albert** [comment]
> slider with a min and max range

@Bandit I didn't understand what you meant by this, are you talking about a range slider (multiple thumbs) that represents multiple numeric ranges?

**Comment by Bandit** [comment]
Yes a range slider where you can select a single value by dragging both thumbs/handles on top of each other. At that point the UX is not clear which thumb is on top, and therefore extremely confusing to the user why they can't drag in a certain direction.

We've had users report the slider is broken, not realising they have to drag left before they can drag right, or vice versa.

**Comment by atomiks** [comment]
> We've had users report the slider is broken, not realising they have to drag left before they can drag right, or vice versa.

I wonder if swapping via pointer usage could be allowed, while disallowed for keyboard usage

**Comment by Bandit** [comment]
I started writing my own one, and it seems to work fine using keyboard and tab order, focus management, etc, not sure I completely understand the issue. Really surprises me that 90% or more of the sliders out there get this fundamental UX problem wrong!

https://codesandbox.io/p/sandbox/sharp-cray-zm62vd
